packages/common/src/intl/intlHelpers.ts

```diff
@@ -20,6 +20,10 @@ export interface TypedTFunction<Keys> {
   ): string;
 }
 
+export const isSupportedLang = (lang: string): boolean => {
+  return lang === 'en' || lang === 'fr' || lang === 'ar';
```

Comment:
Is there a way to request from the server what languages are supported? Ideally these wouldn't be hardcoded. 

Do you think it's worth making a request to try and load the selected language and then defaulting to "en" if that query fails?

Reply:
> Is there a way to request from the server what languages are supported? Ideally these wouldn't be hardcoded.

Nah, there isn't. 

At the moment we'd still need to commit the language files to the repo, so, avoiding hard coding doesn't give you much. Language support is also rare enough [at the moment] that creating a more dynamic system would cost much more than a little bit of hard coded updates.

> Do you think it's worth making a request to try and load the selected language and then defaulting to "en" if that query fails?

Soz - I'm not getting this one. Request with what to where? I thought maybe you were saying a request to fetch the default language from the server. That query isn't supported, so no point sending one. 

Though I did think setting up for the future was a good idea so I have made a hook `useDefaultLanguage` which can be modified in the future to make a query

Reply:
> Soz - I'm not getting this one. Request with what to where? I thought maybe you were saying a request to fetch the default language from the server. That query isn't supported, so no point sending one.

Sorry, my misunderstanding of where the language strings are coming from. I think what I was saying is, is it possible, instead of having hard-coded "supported" languages, can you try and load whatever language and then switch to default "en" if that fails.

```
const { language } = i18n;
    if (!isSupportedLang(language)) i18n.changeLanguage(defaultLanguage);
```

What happens if you do `i18n.changeLanguage(language)` on a language that is not supported? Do you get an error? If so, what about:
```
try {
  i18n.changeLanguage(language)
} catch {
  i18n.changeLanguage(defaultLanguage)
}

Or something to that effect?

Reply:
doesn't throw

we know exactly the languages supported so we can be explicit

Reply:
How about unifying with `SupportedLocales`?

```
const locales = ['en', 'ar', 'fr'];

export type SupportedLocales = typeof locales[number];
export const isSupportedLang = (lang: string): boolean =>
  locales.some(locale => lang === locale);
```

then the list is just in one place 😁

Reply:
sweet. Added some `const`s  - `typeof locales[number]` just resolves to a string